File: bindings/js/examples/hello.ts
// TypeScript hello-world example using the JS/TS binding.
//
//   npm install
//   npm run build
//   npx ts-node examples/hello.ts

import { Unique } from '..';

const app = new Unique();

app.get('/hello', (_req, res) => {
  res.json({ message: 'world', framework: 'unique', lang: 'typescript' });
});

app.post('/echo/:name', async (req, res) => {
  res.json({
    hello: req.params.name,
    youSent: req.body,
  });
});

app.listen(3000).then(() => console.log('🥋 unique listening on http://localhost:3000'));

File: bindings/python/examples/hello.py
"""Hello-world example using the Unique Python binding.

Run with:
    pip install maturin
    maturin develop --release
    python examples/hello.py
"""

from unique import Unique

app = Unique()


@app.get("/hello")
def hello(req):
    return {
        "status": 200,
        "headers": {},
        "body": {"message": "world", "framework": "unique", "lang": "python"},
    }


@app.post("/echo/:name")
async def echo(req):
    return {
        "status": 200,
        "headers": {},
        "body": {"hello": req["params"]["name"], "you_sent": req["body"]},
    }


@app.get("/")
def index(req):
    return {
        "status": 200,
        "headers": {"content-type": "text/html"},
        "body": "<h1>Hello from Unique (Python)!</h1><p>Try /hello or POST /echo/yourname</p>",
    }


if __name__ == "__main__":
    print("🥋 Unique (Python) listening on http://localhost:3000")
    app.run(port=3000)
